Error in code generation phase: reference to unexisting field

Hello,

it looks like that selma get confused when field's name ends with the name of the containing class.

When executed on the following classes:

```
public class Source {
    private String name;
    public String getName() { return name;  }
    public void setName(String field) { this.name = field; }
}

public class Target {
    public static class Bean {
        private String name;
        public String getName() { return name; }
        public void setName(String field) { this.name = field; }
    }

    // Field name suffix is equal to class name
    private Bean fieldTarget;
    public Bean getFieldTarget() { return fieldTarget; }
    public void setFieldTarget(Bean fieldTarget) { this.fieldTarget = fieldTarget; }
}

@Mapper
public interface BeanMapper {
    @Maps( withCustomFields = @Field({"name", "fieldTarget.name"}))
    Target toBean(Source b);
}
```

generate the following error:

```
[ERROR] diagnostic: ..../selma-test/src/main/java/test/selma/BeanMapper.java:11: error: Mapping custom field name from source bean test.selma.Source, setter for field fieldname is missing in destination bean test.selma.Target !
    Target toBean(Source b);
           ^
   --> Check your custom field @Field{ "fieldtarget.name", "name"}, you can add simple class name or FQCN to the longest field path  or add missing getter
```

--- a/processor/src/main/java/fr/xebia/extras/selma/codegen/FieldMap.java
+++ b/processor/src/main/java/fr/xebia/extras/selma/codegen/FieldMap.java
@@ -130,7 +130,7 @@ class Field {
 
     private String removePrefixes(String key, String simpleName, String fqcn) {
         key = key.replace(simpleName + ".", "");
-        return key.replace(fqcn + ".", "");
+        return key.replaceFirst("^" + fqcn + "\\.", "");
     }
 
     public boolean hasEmbedded() {

--- a/processor/src/test/java/fr/xebia/extras/selma/beans/Home.java
+++ b/processor/src/test/java/fr/xebia/extras/selma/beans/Home.java
@@ -1,0 +1,22 @@
+package fr.xebia.extras.selma.beans;
+
+/**
+ * @author FaniloRandria
+ */
+public class Home {
+
+	private Phone phoneHome;
+	
+	public Home (Phone phoneHome) {
+		this.phoneHome = phoneHome;
+	}
+
+	public Phone getPhoneHome() {
+		return phoneHome;
+	}
+
+	public void setPhoneHome(Phone phoneHome) {
+		this.phoneHome = phoneHome;
+	}
+	
+}

--- a/processor/src/test/java/fr/xebia/extras/selma/beans/HomeDto.java
+++ b/processor/src/test/java/fr/xebia/extras/selma/beans/HomeDto.java
@@ -1,0 +1,18 @@
+package fr.xebia.extras.selma.beans;
+
+/**
+ * @author FaniloRandria
+ */
+public class HomeDto {
+
+	public String phone;
+
+	public String getPhone() {
+		return phone;
+	}
+
+	public void setPhone(String phone) {
+		this.phone = phone;
+	}
+
+}

--- a/processor/src/test/java/fr/xebia/extras/selma/beans/Phone.java
+++ b/processor/src/test/java/fr/xebia/extras/selma/beans/Phone.java
@@ -1,0 +1,22 @@
+package fr.xebia.extras.selma.beans;
+
+/**
+ * @author FaniloRandria
+ */
+public class Phone {
+	
+	private String phoneNumber;
+	
+	public Phone (String phoneNumber) {
+		this.phoneNumber = phoneNumber;
+	}
+
+	public String getPhoneNumber() {
+		return phoneNumber;
+	}
+
+	public void setPhoneNumber(String phoneNumber) {
+		this.phoneNumber = phoneNumber;
+	}
+
+}

--- a/processor/src/test/java/fr/xebia/extras/selma/it/HomeMapperIT.java
+++ b/processor/src/test/java/fr/xebia/extras/selma/it/HomeMapperIT.java
@@ -1,0 +1,32 @@
+package fr.xebia.extras.selma.it;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+
+import fr.xebia.extras.selma.Selma;
+import fr.xebia.extras.selma.beans.Home;
+import fr.xebia.extras.selma.beans.HomeDto;
+import fr.xebia.extras.selma.beans.Phone;
+import fr.xebia.extras.selma.it.mappers.HomeMapper;
+import fr.xebia.extras.selma.it.utils.Compile;
+import fr.xebia.extras.selma.it.utils.IntegrationTestBase;
+
+/**
+ * @author FaniloRandria
+ */
+@Compile(withClasses = HomeMapper.class)
+public class HomeMapperIT extends IntegrationTestBase {
+
+    public static final String NUMBER_PHONE = "06-11-22-33-44";
+
+    @Test
+    public void beanMapperTest() throws Exception {
+        HomeMapper mapper = Selma.getMapper(HomeMapper.class);
+        Phone phone = new Phone(NUMBER_PHONE);
+        Home home = new Home(phone);
+        HomeDto homeDto = mapper.toDto(home);
+        assertThat(homeDto.getPhone()).isEqualTo(NUMBER_PHONE);
+    }
+
+}

--- a/processor/src/test/java/fr/xebia/extras/selma/it/mappers/HomeMapper.java
+++ b/processor/src/test/java/fr/xebia/extras/selma/it/mappers/HomeMapper.java
@@ -1,0 +1,19 @@
+package fr.xebia.extras.selma.it.mappers;
+
+import fr.xebia.extras.selma.Field;
+import fr.xebia.extras.selma.Mapper;
+import fr.xebia.extras.selma.Maps;
+import fr.xebia.extras.selma.beans.Home;
+import fr.xebia.extras.selma.beans.HomeDto;
+
+/**
+ * @author FaniloRandria
+ */
+@Mapper
+public interface HomeMapper {
+
+    @Maps(withCustomFields = {
+    		@Field(value = {"phone", "phoneHome.phoneNumber"})
+    }) HomeDto toDto (Home in);
+    
+}

